app/src/main/java/com/example/weather/DetailsFragment.kt

```diff
@@ -86,8 +93,53 @@ class DetailsFragment : Fragment() {
         val formatter = DateTimeFormatter.ofPattern("K:mm a", Locale.ENGLISH)
         return instant.atOffset(offset).format(formatter)
     }
+
+
+    private fun displayDetailsError() {
+        binding.name.text = EMPTY_STRING
+        binding.country.text = EMPTY_STRING
+        binding.temp.text = EMPTY_STRING
+        binding.lowAndHigh.text = EMPTY_STRING
+        binding.humidity.text = EMPTY_STRING
+        binding.windspeedAmount.text = EMPTY_STRING
+        binding.pressureAmount.text = EMPTY_STRING
+        binding.description.text = EMPTY_STRING
+        binding.sunrise.text = EMPTY_STRING
+        binding.sunset.text = EMPTY_STRING
+        binding.humidityIcon.visibility = View.GONE
+        binding.loading.visibility = View.GONE
+        binding.firstLine.visibility = View.GONE
+        binding.secondLine.visibility = View.GONE
+        binding.snag.visibility = View.VISIBLE
+        binding.errorMessage.visibility = View.VISIBLE
+        binding.errorMessage.text = getString(R.string.error_message)
```

Comment:
Since this error string is always the same, it might be easier to set the string's value in the xml file so you don't need to set the string everytime you wanna display the error